### PR TITLE
Remove dataset from thumbnail cache

### LINKED ISSUE #3808: Reload thumbnail on dataset reload

### PR DESCRIPTION
**Pull request by fm3**
### URL of deployed dev instance (used for testing):
- https://removefromthumbnailcache.webknossos.xyz

### Steps to test:
- new route `PUT /datasets/:organizationName/:dataSetName/clearThumbnailCache` (no body expected)
- (should be triggered by the reload button in the dataset list view)
- after that, a hard page refresh in the gallery should trigger new thumbnail requests from the wK backend to the datastore

### Issues:
- fixes #3808

------
- ~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

File: app/controllers/DataSetController.scala [hunk 1]
```diff
@@ -48,12 +47,22 @@ class DataSetController @Inject()(userService: UserService,
       (__ \ 'sortingKey).readNullable[Long] and
       (__ \ 'isPublic).read[Boolean]).tupled
 
+  def removeFromThumbnailCache(organizationName: String, dataSetName: String) = {
+    sil.SecuredAction { implicit request =>
+      dataSetService.thumbnailCache.removeAllConditional(key => key.startsWith(s"thumbnail-$organizationName*$dataSetName"))
```
Comment (youri-k) [inline app/controllers/DataSetController.scala]: I think you can shorten this to `_.startsWith(..)`

File: frontend/javascripts/dashboard/advanced_dataset/dataset_action_view.js [hunk 2]
```diff
@@ -32,6 +36,7 @@ class DatasetActionView extends React.PureComponent<Props, State> {
 
   clearCache = async (dataset: APIMaybeUnimportedDataset) => {
     await triggerDatasetClearCache(dataset.dataStore.url, dataset);
+    await triggerDatasetClearThumbnailCache(dataset);
```
Comment (daniel-wer) [inline frontend/javascripts/dashboard/advanced_dataset/dataset_action_view.js]: @fm3 Does the backend expect these requests in order? Otherwise you could use Promise.all to execute them concurrently :)

```javascript
await Promise.all([
  triggerDatasetClearCache(dataset.dataStore.url, dataset),
  triggerDatasetClearThumbnailCache(dataset),
]);
```
